Framework/DataHandling/src/GenerateGroupingPowder.cpp

```diff
@@ -48,16 +115,34 @@ const std::string GenerateGroupingPowder::category() const {
 /** Initialize the algorithm's properties.
  */
 void GenerateGroupingPowder::init() {
-  declareProperty(std::make_unique<WorkspaceProperty<>>("InputWorkspace", "", Direction::Input),
+  declareProperty(std::make_unique<WorkspaceProperty<>>("InputWorkspace", "", Direction::Input,
+                                                        std::make_shared<InstrumentValidator>()),
                   "A workspace from which to generate the grouping.");
+
+  // allow saving as either xml or hdf5 formats
+  auto fileExtensionXMLorHDF5 = std::make_shared<ListValidator<std::string>>();
+  fileExtensionXMLorHDF5->addAllowedValue(std::string("xml"));
+  fileExtensionXMLorHDF5->addAllowedValue(std::string("nxs"));
+  fileExtensionXMLorHDF5->addAllowedValue(std::string("nx5"));
+  declareProperty("FileFormat", std::string("xml"), fileExtensionXMLorHDF5,
+                  "File extension/format for saving output: either xml (default) or nxs/nx5.");
+
   auto positiveDouble = std::make_shared<BoundedValidator<double>>();
   positiveDouble->setLower(0.0);
   declareProperty("AngleStep", -1.0, positiveDouble, "The angle step for grouping, in degrees.");
-  declareProperty(std::make_unique<FileProperty>("GroupingFilename", "", FileProperty::Save, ".xml"),
+  declareProperty("AzimuthalStep", -1.0, positiveDouble, "The azimuthal angle step for grouping, in degrees.");
```

Comment:
The default should be 360 (basically ignore everything) and the range should be (0, 360] (exclusive, inclusive]